llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3679,17 +3679,38 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpGroupNonUniformShuffleDow
 }
 
 // =====================================================================================================================
-// Handle OpTraceRayKHR.
+// Create TraceRay dialect Op.
 //
 // @param spvValue : A SPIR-V value.
-template <> Value *SPIRVToLLVM::transValueWithOpcode<OpTraceRayKHR>(SPIRVValue *const spvValue) {
+Value *SPIRV::SPIRVToLLVM::createTraceRayDialectOp(SPIRVValue *const spvValue) {
   if (m_execModule != ExecutionModelRayGenerationKHR) {
     Llpc::Context *llpcContext = static_cast<Llpc::Context *>(m_context);
     auto *pipelineContext = static_cast<Llpc::RayTracingContext *>(llpcContext->getPipelineContext());
     pipelineContext->setIndirectPipeline();
   }
+
+  SPIRVInstruction *const spvInst = static_cast<SPIRVInstruction *>(spvValue);
+  std::vector<SPIRVValue *> spvOperands = spvInst->getOperands();
   BasicBlock *const block = getBuilder()->GetInsertBlock();
-  return mapValue(spvValue, transSPIRVBuiltinFromInst(static_cast<SPIRVInstruction *>(spvValue), block));
+  Function *const func = getBuilder()->GetInsertBlock()->getParent();
```

Comment:
You'd reuse the scoped enum from above here

Reply:
I'd rather keep them separate because the internal-only opcode will have a slightly different index order than the dialect.

Reply:
Index is used every elsewhere in SPIRVReader, and I think the names of the variable get assigned are already representative enough?

Reply:
Just my personal taste - just leave it as it is :)